src/test/java/teammates/logic/core/FeedbackResponseCommentsLogicTest.java

```diff
@@ -346,6 +353,144 @@ public void testGetFeedbackResponseCommentsForQuestionInSection_withSectionName_
+
+        ______TS("success: giver is student; show comment to student; comment is visible to recipient");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student1InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is student; show comment to student; comment is visible to teammates");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student3InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        relatedComment = FeedbackResponseCommentAttributes.builder()
+                .withCommentGiver("student4InCourse1@gmail.tmt")
+                .withCommentGiverType(FeedbackParticipantType.STUDENTS)
+                .withVisibilityFollowingFeedbackQuestion(false)
+                .withCourseId("idOfTypicalCourse1")
+                .withFeedbackSessionName("First feedback session")
+                .withShowCommentTo(Collections.emptyList())
+                .build();
+
+        ______TS("success: giver is student; showCommentTo is empty; comment is visible to giver");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student4InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is student; showCommentTo is empty; comment is not visible to recipient");
+        assertFalse(frcLogic.isResponseCommentVisibleForUser("student1InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is student; showCommentTo is empty; comment is not visible to teammates");
+        assertFalse(frcLogic.isResponseCommentVisibleForUser("student3InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is student; showCommentTo is empty; comment is not visible to instructor");
+        assertFalse(frcLogic.isResponseCommentVisibleForUser("instructor1@course1.tmt", UserRole.INSTRUCTOR,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        relatedComment = FeedbackResponseCommentAttributes.builder()
+                .withCommentGiver("instructor1@course1.tmt")
+                .withCommentGiverType(FeedbackParticipantType.INSTRUCTORS)
+                .withVisibilityFollowingFeedbackQuestion(false)
+                .withCourseId("idOfTypicalCourse1")
+                .withFeedbackSessionName("First feedback session")
+                .withShowCommentTo(Collections.emptyList())
+                .build();
+
+        ______TS("success: giver is instructor; showCommentTo is empty; comment is visible to giver by default");
```

Comment:
nit: Consider standardising the phrasing, esp for the centre part to something like `do not show comment to everyone except self`. At present, this part looks slightly out of place.

Reply:
Thanks for the suggestion. It is updated now.